packages/x-date-pickers/src/hooks/useClearableField.tsx

```diff
@@ -76,6 +76,8 @@ export const useClearableField = <TFieldProps extends UseClearableFieldProps>(
 
   return {
     ...other,
+    slots,
+    slotProps,
```

Comment:
It makes this test fail: https://app.circleci.com/pipelines/github/mui/mui-x/64286/workflows/bd74a85a-9a6b-48ab-b2ab-0d8767672ae2/jobs/366882
Also, slots and slotProps shouldn't be here according to the type https://github.com/mui/mui-x/blob/c911ad501c4961e63e0e5c9b9eacf4e6367d3210/packages/x-date-pickers/src/hooks/useClearableField.tsx#L47-L50

Reply:
cc @LukasTy

Reply:
I've reverted this change to make the tests pass.
Why was it necessary?

Reply:
To properly support the migration to `slots` and `slotProps` on `TextField`.
I'll handle this. 😉

Reply:
Created a separate issue for it as it's not a critical issue and would explode the scope of this PR.